plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XWorkspaceManager.java

```diff
@@ -313,6 +314,12 @@ public XProjectManager getProjectManager(URI uri) {
 		String name = null;
 		if (projectConfig != null) {
 			name = projectConfig.getName();
+		} else if (N4Scheme.isN4Scheme(uri)) {
```

Comment:
The intention was to keep X-classes free of N4JS related logic.

Reply:
Ok, I'll extract a subclass in a subsequent step